src/main/java/org/codehaus/mojo/versions/api/DefaultVersionsHelper.java

```diff
@@ -806,15 +771,10 @@ public ArtifactVersions lookupDependencyUpdates( Dependency dependency, boolean
                 pluginUpdates.put( pud.getPlugin(), pud.getPluginUpdatesDetails() );
             }
         }
-        catch ( final ExecutionException ee )
-        {
-            throw new ArtifactMetadataRetrievalException( "Unable to acquire metadata for plugins " + plugins + ": "
-                + ee.getMessage(), ee );
-        }
-        catch ( final InterruptedException ie )
+        catch ( ExecutionException | InterruptedException ie )
         {
             throw new ArtifactMetadataRetrievalException( "Unable to acquire metadata for plugins " + plugins + ": "
-                + ie.getMessage(), ie );
+                + ie.getMessage(), ie, null );
```

Comment:
Same here, what do you think of extracting the artifact into a variable outside the try statement, so we can pass it here? Also change to the new legacy package.

Reply:
same please looks at the context